app/controllers/projects/settings/life_cycles_controller.rb

```diff
@@ -0,0 +1,114 @@
+# -- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) 2010-2024 the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+# ++
+
+class Projects::Settings::LifeCyclesController < Projects::SettingsController
+  include OpTurbo::ComponentStream
+
+  before_action :deny_access_on_feature_flag
+
+  before_action :load_life_cycle_definitions, only: %i[show enable_all disable_all]
+
+  menu_item :settings_life_cycles
+
+  def show; end
+
+  def toggle
+    step_params = params
+                    .require(:project_life_cycle)
+                    .permit(:definition_id, :type)
+                    .to_h
+                    .symbolize_keys
+                    .merge(active: params["value"] == "1")
+
+    upsert_one_step(**step_params)
+  end
+
+  def disable_all
+    upsert_all_steps(active: false)
+
+    redirect_to action: :show
+  end
+
+  def enable_all
+    upsert_all_steps(active: true)
+
+    redirect_to action: :show
+  end
+
+  private
+
+  def load_life_cycle_definitions
+    @life_cycle_definitions = Project::LifeCycleStepDefinition.order(position: :asc)
```

Comment:
I've added `default_scope { order(:position) }` in the PR for global stage admin

Reply:
This is probably a question of believe as there are good arguments and bad for/against using default scopes. I am aware that e.g. in meetings, the default order is currently being removed. In my book a default order is not as bad as other default scopes which can seriously mess up a day. But here I am not as decided.

My argument for keeping an explicit order basically boils down to:
* It is explicit. So when you read the code, you are not surprised as to what happens.

But I would be open to removing it as well (in the other PR).

Reply:
I would like to leave it in for now then. Once the other PR is moved, I would be ok with removing the order here (I still lean towards leaving it). But currently, it would need to be kept in here to ensure the order.